homework-submission/todo-app/src/todos/addform.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import DatePicker from 'react-datepicker';
+import moment from 'moment';
+ 
+import 'react-datepicker/dist/react-datepicker.css';
+import {inject, observer} from 'mobx-react';
+
+@inject('TodosStore')
+@observer
+
+class SendForm extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            description: '',
+            deadline: '',
+            done: false,
+            startDate: moment()
+        };
+    }
+
+    handleDateChange = date => {
+        this.setState({
+            deadline: date.format("DD-MM-YYYY"),
+            startDate: date
+        });
+    }
+
+    handleTextChange = e => {
+        this.setState({[e.target.name]: e.target.value});
+    }
+
+    handleSubmit = e => {
+        const {TodosStore} = this.props;
+        e.preventDefault();
+        const checkDateField = this.state.deadline;
+        if(checkDateField ==='') {
+            const itemOfToday = {
+                ...this.state,
+                deadline: moment().format("DD-MM-YYYY")
+            }
+            TodosStore.addNewItem(itemOfToday);
+        }else TodosStore.addNewItem(this.state);
+        
+        const initialState = {
```

Comment:
Great initializing ! 🎉
---------------------
And this would be more use-able out of the class and initialize the original state by it 👌